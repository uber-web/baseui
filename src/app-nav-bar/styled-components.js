/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled, withStyle} from '../styles/index.js';
import {StyledListItem} from '../menu/index.js';

const StyledClearButton = styled<{$isFocusVisible: boolean}>(
  'button',
  ({$theme, $isFocusVisible}) => ({
    boxSizing: 'border-box',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
    color: $theme.colors.contentPrimary,
    borderLeftWidth: 0,
    borderTopWidth: 0,
    borderRightWidth: 0,
    borderBottomWidth: 0,
    paddingTop: '0',
    paddingBottom: '0',
    paddingLeft: '0',
    paddingRight: '0',
    marginLeft: 0,
    marginTop: 0,
    marginRight: 0,
    marginBottom: 0,
    outline: $isFocusVisible ? `3px solid ${$theme.colors.accent}` : 'none',
    outlineOffset: '-3px',
    WebkitAppearance: 'none',
    cursor: 'pointer',
  }),
);

export const StyledRoot = styled<{}>('div', props => {
  const {$theme} = props;
  return {
    ...$theme.typography.font300,
    boxSizing: 'border-box',
    backgroundColor: $theme.colors.backgroundPrimary,
    boxShadow: $theme.lighting.shadow600,
    width: '100%',
  };
});

export const StyledSpacing = styled<{}>('div', props => {
  const {$theme} = props;
  return {
    boxSizing: 'border-box',
    height: '100%',
    paddingTop: $theme.sizing.scale700,
    paddingBottom: $theme.sizing.scale700,
    display: 'flex',
    alignItems: 'center',
    [`@media screen and (max-width: ${$theme.breakpoints.medium - 1}px)`]: {
      paddingTop: $theme.sizing.scale400,
      paddingBottom: $theme.sizing.scale400,
    },
  };
});

export const StyledAppName = styled<{}>('span', ({$theme}) => ({
  ...$theme.typography.font650,
  color: $theme.colors.primary,
  textDecoration: 'none',
  [`@media screen and (max-width: ${$theme.breakpoints.medium - 1}px)`]: {
    ...$theme.typography.font550,
  },
}));

// $FlowFixMe
export const StyledSideMenuButton = withStyle<{}>(
  StyledClearButton,
  ({$theme}) => ({
    marginRight: $theme.sizing.scale600,
    paddingTop: $theme.sizing.scale100,
    paddingBottom: $theme.sizing.scale100,
    paddingLeft: $theme.sizing.scale100,
    paddingRight: $theme.sizing.scale100,
  }),
);

export const StyledPrimaryMenuContainer = styled<{}>('div', ({$theme}) => {
  return {
    boxSizing: 'border-box',
    height: '100%',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'flex-end',
    alignItems: 'stretch',
  };
});

export const StyledPrimaryMenuItem = styled<{
  $active: boolean,
  $isFocusVisible: boolean,
}>('div', props => {
  const {
    $active,
    $isFocusVisible,
    $theme: {colors, sizing},
  } = props;
  return {
    boxSizing: 'border-box',
    display: 'flex',
    alignItems: 'center',
    color: $active ? colors.contentPrimary : colors.contentTertiary,
    marginLeft: sizing.scale700,
    marginRight: sizing.scale700,
    outline: $isFocusVisible ? `3px solid ${colors.accent}` : 'none',
    outlineOffset: '-3px',
    borderBottom:
      $active && !$isFocusVisible
        ? `2px solid ${colors.primary}`
        : '2px solid transparent',
    ':hover': {
      color: colors.primary,
    },
  };
});

export const StyledUserMenuButton = StyledClearButton;

export const StyledUserMenuListItem = withStyle(StyledListItem, {
  paddingTop: '0',
  paddingBottom: '0',
  paddingRight: '0',
});

export const StyledUserProfileTileContainer = styled<{}>('div', ({$theme}) => {
  return {
    boxSizing: 'border-box',
    height: '100%',
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'flex-start',
    paddingTop: $theme.sizing.scale650,
    paddingBottom: $theme.sizing.scale650,
  };
});

export const StyledUserProfilePictureContainer = styled<{}>(
  'div',
  ({$theme}) => {
    return {
      marginRight: $theme.sizing.scale600,
    };
  },
);

export const StyledUserProfileInfoContainer = styled<{}>('div', ({$theme}) => {
  return {
    boxSizing: 'border-box',
    alignSelf: 'center',
  };
});
