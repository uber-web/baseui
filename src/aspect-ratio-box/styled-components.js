/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled} from '../styles/index.js';

export const Root = styled<{$aspectRatio: number}>('div', ({$aspectRatio}) => ({
  width: '100%',
  position: 'relative',
  '::before': {
    content: '""',
    width: '1px',
    marginLeft: '-1px',
    float: 'left',
    height: 0,
    paddingTop: `${100 / $aspectRatio}%`,
    pointerEvents: 'none',
  },
  '::after': {
    content: '""',
    display: 'table',
    clear: 'both',
  },
}));

export const Body = styled<{}>('div', () => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  width: '100%',
}));
