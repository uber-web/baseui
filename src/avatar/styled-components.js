/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import {styled} from '../styles/index.js';
import type {StylePropsT} from './types.js';

function getSize(props: StylePropsT) {
  const {$size, $theme} = props;

  const defaultSize = $theme.sizing.scale1000;
  const size = $size || defaultSize;
  return $theme.sizing[size] || size;
}

export const Avatar = styled('img', (props: StylePropsT) => {
  const themedSize = getSize(props);

  return {
    borderTopRightRadius: '50%',
    borderBottomRightRadius: '50%',
    borderTopLeftRadius: '50%',
    borderBottomLeftRadius: '50%',
    boxSizing: 'border-box',
    display: 'block',
    height: themedSize,
    width: themedSize,
    objectFit: 'cover',
  };
});

export const Initials = styled('div', (props: StylePropsT) => ({
  ...props.$theme.typography.font400,
  color: props.$theme.colors.mono100,
  alignItems: 'center',
  display: 'flex',
  justifyContent: 'center',
  height: '100%',
}));

export const Root = styled('div', (props: StylePropsT) => {
  const {$didImageFailToLoad} = props;
  const themedSize = getSize(props);

  return {
    backgroundColor: $didImageFailToLoad ? props.$theme.colors.primary : null,
    borderTopRightRadius: '50%',
    borderBottomRightRadius: '50%',
    borderTopLeftRadius: '50%',
    borderBottomLeftRadius: '50%',
    boxSizing: 'border-box',
    display: 'inline-block',

    // image previously set the root height/width
    // since image is not rendered, set the height/width
    height: $didImageFailToLoad ? themedSize : null,
    width: $didImageFailToLoad ? themedSize : null,
  };
});
