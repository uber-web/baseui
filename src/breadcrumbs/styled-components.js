/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import {styled} from '../styles/index.js';

export const StyledRoot = styled<{}>('nav', ({$theme}) => {
  return {
    color: $theme.colors.breadcrumbsText,
    ...$theme.typography.font450,
  };
});

export const StyledSeparator = styled<{}>('div', ({$theme}) => {
  return {
    display: 'inline-block',
    color: $theme.colors.breadcrumbsSeparatorFill,
    marginLeft: $theme.sizing.scale300,
    marginRight: $theme.sizing.scale300,
  };
});
