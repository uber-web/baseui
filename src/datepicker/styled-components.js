/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import {styled} from '../styles/index.js';
import type {SharedStylePropsT} from './types.js';

/**
 * Main component container element
 */
export const StyledRoot = styled('div', (props: SharedStylePropsT) => {
  const {
    $theme: {typography, colors, borders},
  } = props;
  const borderRadius = borders.useRoundedCorners ? borders.radius200 : '0px';
  return {
    ...typography.font400,
    color: props.$theme.colors.datepickerDayFont,
    backgroundColor: colors.datepickerBackground,
    textAlign: 'center',
    borderTopRightRadius: borderRadius,
    borderBottomRightRadius: borderRadius,
    borderTopLeftRadius: borderRadius,
    borderBottomLeftRadius: borderRadius,
    display: 'inline-block',
  };
});

export const StyledCalendarContainer = styled(
  'div',
  (props: SharedStylePropsT) => {
    const {
      $theme: {sizing},
    } = props;
    return {
      paddingTop: sizing.scale400,
      paddingBottom: sizing.scale500,
      paddingLeft: sizing.scale600,
      paddingRight: sizing.scale600,
    };
  },
);

export const StyledQuickSelectContainer = styled(
  'div',
  (props: SharedStylePropsT) => ({
    maxWidth: '296px',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginBottom: props.$theme.sizing.scale600,
  }),
);

export const StyledQuickSelectLabel = styled(
  'div',
  (props: SharedStylePropsT) => ({
    ...props.$theme.typography.font300,
    color: props.$theme.colors.colorPrimary,
    marginBottom: props.$theme.sizing.scale600,
    textAlign: 'left',
  }),
);

export const StyledQuickSelectButtons = styled('div', {
  display: 'flex',
  flexWrap: 'wrap',
});

export const StyledCalendarHeader = styled(
  'div',
  (props: SharedStylePropsT) => {
    const {
      $theme: {colors, sizing, borders},
    } = props;
    const borderRadius = borders.useRoundedCorners ? borders.radius200 : '0px';
    return {
      color: colors.white,
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingTop: sizing.scale500,
      paddingBottom: sizing.scale500,
      paddingLeft: sizing.scale600,
      paddingRight: sizing.scale600,
      backgroundColor: colors.primary,
      borderTopRightRadius: borderRadius,
      borderBottomRightRadius: 0,
      borderTopLeftRadius: borderRadius,
      borderBottomLeftRadius: 0,
    };
  },
);

export const StyledMonthHeader = styled('div', (props: SharedStylePropsT) => {
  return {
    whiteSpace: 'no-wrap',
  };
});

function getArrowBtnStyle({$theme}: SharedStylePropsT) {
  return {
    boxSizing: 'border-box',
    height: '22px',
    color: $theme.colors.white,
    backgroundColor: 'transparent',
    borderWidth: '0',
    paddingTop: '3px',
    paddingBottom: '3px',
    paddingLeft: '3px',
    paddingRight: '3px',
    outline: 'none',
    ':focus': {
      backgroundColor: $theme.colors.primary500,
      borderRadius: $theme.borders.useRoundedCorners
        ? $theme.sizing.scale100
        : 0,
    },
  };
}

export const StyledPrevButton = styled('button', getArrowBtnStyle);

export const StyledNextButton = styled('button', getArrowBtnStyle);

export const StyledMonth = styled('div', (props: SharedStylePropsT) => {
  return {
    display: 'inline-block',
  };
});

export const StyledWeek = styled('div', (props: SharedStylePropsT) => {
  const {
    $theme: {sizing},
  } = props;
  return {
    whiteSpace: 'no-wrap',
    display: 'flex',
    marginBottom: sizing.scale100,
  };
});

type BorderRadiusT = {
  borderTopLeftRadius: string | number,
  borderBottomLeftRadius: string | number,
  borderTopRightRadius: string | number,
  borderBottomRightRadius: string | number,
};

function getBorderRadius(left, right): BorderRadiusT {
  return {
    borderTopLeftRadius: left,
    borderBottomLeftRadius: left,
    borderTopRightRadius: right,
    borderBottomRightRadius: right,
  };
}

export function calculateBorderRadius(
  props: SharedStylePropsT,
): ?BorderRadiusT {
  const {
    $isHighlighted,
    $pseudoHighlighted,
    $pseudoSelected,
    $selected,
    $startDate,
    $range,
    $hasRangeHighlighted,
    $hasRangeOnRight,
    $hasRangeSelected,
    $theme: {borders},
  } = props;
  if (borders.useRoundedCorners) {
    if ($selected) {
      if (!$range) {
        return getBorderRadius(borders.radius200, borders.radius200);
      } else {
        if ($hasRangeSelected) {
          return $startDate
            ? getBorderRadius(borders.radius200, 0)
            : getBorderRadius(0, borders.radius200);
        } else {
          if ($hasRangeHighlighted) {
            return $hasRangeOnRight
              ? getBorderRadius(borders.radius200, 0)
              : getBorderRadius(0, borders.radius200);
          } else {
            return getBorderRadius(borders.radius200, borders.radius200);
          }
        }
      }
    } else {
      if (!$isHighlighted && ($pseudoHighlighted || $pseudoSelected)) {
        return getBorderRadius(0, 0);
      } else {
        if ($isHighlighted) {
          if (!$range) {
            return getBorderRadius(borders.radius200, borders.radius200);
          } else if ($hasRangeHighlighted) {
            return $hasRangeOnRight
              ? getBorderRadius(0, borders.radius200)
              : getBorderRadius(borders.radius200, 0);
          } else {
            return $pseudoSelected
              ? getBorderRadius(0, 0)
              : getBorderRadius(borders.radius200, borders.radius200);
          }
        } else {
          return !$pseudoSelected
            ? getBorderRadius(borders.radius200, borders.radius200)
            : null;
        }
      }
    }
  } else {
    return getBorderRadius(0, 0);
  }
}

export const StyledDay = styled('div', (props: SharedStylePropsT) => {
  const {
    $disabled,
    $isHovered,
    $isHighlighted,
    $outsideMonth,
    $pseudoHighlighted,
    $pseudoSelected,
    $selected,
    $theme: {colors, sizing, borders},
  } = props;
  return {
    boxSizing: 'border-box',
    position: 'relative',
    cursor: $disabled ? 'default' : 'pointer',
    display: 'inline-block',
    width: sizing.scale1000,
    height: sizing.scale1000,
    lineHeight: sizing.scale800,
    textAlign: 'center',
    paddingTop: sizing.scale300,
    paddingBottom: sizing.scale300,
    paddingLeft: sizing.scale200,
    paddingRight: sizing.scale200,
    marginTop: '0',
    marginBottom: '0',
    marginLeft: '0',
    marginRight: '0',
    color: $selected
      ? colors.white
      : $outsideMonth || $disabled
        ? colors.datepickerDayFontDisabled
        : 'inherit',
    backgroundColor: $selected
      ? $isHighlighted
        ? colors.primary500
        : colors.primary
      : $pseudoSelected
        ? $isHighlighted
          ? colors.datepickerDayPseudoHighlighted
          : colors.datepickerDayPseudoSelected
        : $isHovered || $isHighlighted || $pseudoHighlighted
          ? colors.datepickerDayPseudoSelected
          : 'transparent',
    ...calculateBorderRadius(props),
    ':first-child': {
      ...(borders.useRoundedCorners
        ? {
            borderTopLeftRadius: borders.radius200,
            borderBottomLeftRadius: borders.radius200,
          }
        : {}),
    },
    ':last-child': {
      ...(borders.useRoundedCorners
        ? {
            borderTopRightRadius: borders.radius200,
            borderBottomRightRadius: borders.radius200,
          }
        : {}),
    },
  };
});
