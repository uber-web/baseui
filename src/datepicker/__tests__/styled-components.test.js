/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledRoot,
  StyledQuickSelectContainer,
  StyledQuickSelectLabel,
  StyledQuickSelectButtons,
  StyledCalendarContainer,
  StyledCalendarHeader,
  StyledMonthHeader,
  StyledMonth,
  StyledWeek,
  StyledDay,
} from '../index.js';
import {calculateBorderRadius as calcRadius} from '../styled-components.js';

const styledComponents = [
  [StyledRoot, 'StyledRoot'],
  [StyledQuickSelectContainer, 'StyledQuickSelectContainer'],
  [StyledQuickSelectLabel, 'StyledQuickSelectLabel'],
  [StyledQuickSelectButtons, 'StyledQuickSelectButtons'],
  [StyledCalendarContainer, 'StyledCalendarContainer'],
  [StyledCalendarHeader, 'StyledCalendarHeader'],
  [StyledMonthHeader, 'StyledMonthHeader'],
  [StyledMonth, 'StyledMonth'],
  [StyledWeek, 'StyledWeek'],
  [StyledDay, 'StyledDay'],
];

const props = [
  ['$disabled'],
  ['$isHighlighted'],
  ['$isHovered'],
  ['$outsideMonth'],
  ['$selected'],
];

describe('Component styled components', () => {
  test.each(styledComponents)('Basic render', (Component, name) => {
    const component = shallow(<Component />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has correct styles`,
    );
  });

  test.each(props)('Prop based styles', prop => {
    const component = shallow(<StyledDay {...{[prop]: true}} />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `StyledDay has correct styles when ${prop} is set to true`,
    );
  });
});

describe('Border radius calculation', () => {
  function getProps(overrides: any) {
    return {
      $date: new Date(),
      $isHighlighted: false,
      $pseudoHighlighted: false,
      $pseudoSelected: false,
      $selected: false,
      $startDate: false,
      $range: true,
      $hasRangeHighlighted: false,
      $hasRangeOnRight: false,
      $hasRangeSelected: false,
      $theme: {
        borders: {
          useRoundedCorners: true,
          radius200: '4px',
        },
      },
      ...overrides,
    };
  }

  type BorderRadiusT = {
    borderTopLeftRadius: string | number,
    borderBottomLeftRadius: string | number,
    borderTopRightRadius: string | number,
    borderBottomRightRadius: string | number,
  };

  function toStrVal(obj: ?BorderRadiusT) {
    return obj
      ? `${obj.borderTopLeftRadius} ${obj.borderTopRightRadius} ${
          obj.borderBottomRightRadius
        } ${obj.borderBottomLeftRadius}`
      : null;
  }

  test('range and useRoundedCorners', () => {
    expect(toStrVal(calcRadius(getProps()))).toEqual('4px 4px 4px 4px');
  });
  test('!useRoundedCorners', () => {
    // !useRoundedCorners
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $theme: {
              borders: {useRoundedCorners: false, radius200: '4px'},
            },
          }),
        ),
      ),
    ).toEqual('0 0 0 0');
  });
  test('$selected', () => {
    // $selected
    expect(toStrVal(calcRadius(getProps({$selected: true})))).toEqual(
      '4px 4px 4px 4px',
    );
  });
  test('$selected and !range', () => {
    // $selected/!$range
    expect(
      toStrVal(calcRadius(getProps({$selected: true, $range: false}))),
    ).toEqual('4px 4px 4px 4px');
  });
  test('$selected, $hasRangeSelected, and $startDate', () => {
    // $selected/$hasRangeSelected/$startDate
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $selected: true,
            $hasRangeSelected: true,
            $startDate: true,
          }),
        ),
      ),
    ).toEqual('4px 0 0 4px');
  });
  test('$selected, $hasRangeSelected, and !$startDate (endDate)', () => {
    // $selected/$hasRangeSelected/!$startDate(means endDate)
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $selected: true,
            $hasRangeSelected: true,
          }),
        ),
      ),
    ).toEqual('0 4px 4px 0');
  });
  test('$selected, $hasRangeHighlighted, and $hasRangeOnRight', () => {
    // $selected/$hasRangeHighlighted/$hasRangeOnRight
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $selected: true,
            $hasRangeHighlighted: true,
            $hasRangeOnRight: true,
          }),
        ),
      ),
    ).toEqual('4px 0 0 4px');
  });
  test('$selected, $hasRangeHighlighted, and !$hasRangeOnRight(heighlighted range is on the left)', () => {
    // $selected/$hasRangeHighlighted/!$hasRangeOnRight(means heighlighted range is on the left)
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $selected: true,
            $hasRangeHighlighted: true,
          }),
        ),
      ),
    ).toEqual('0 4px 4px 0');
  });
  test('$pseudoHighlighted', () => {
    expect(toStrVal(calcRadius(getProps({$pseudoHighlighted: true})))).toEqual(
      '0 0 0 0',
    );
  });
  test('$pseudoSelected', () => {
    expect(toStrVal(calcRadius(getProps({$pseudoSelected: true})))).toEqual(
      '0 0 0 0',
    );
  });
  test('$isHighlighted and !$range', () => {
    expect(
      toStrVal(calcRadius(getProps({$isHighlighted: true, $range: false}))),
    ).toEqual('4px 4px 4px 4px');
  });
  test('$isHighlighted and $hasRangeHighlighted', () => {
    expect(
      toStrVal(
        calcRadius(
          getProps({$isHighlighted: true, $hasRangeHighlighted: true}),
        ),
      ),
    ).toEqual('4px 0 0 4px');
  });
  test('$isHighlighted, $hasRangeHighlighted, and $hasRangeOnRight', () => {
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $isHighlighted: true,
            $hasRangeHighlighted: true,
            $hasRangeOnRight: true,
          }),
        ),
      ),
    ).toEqual('0 4px 4px 0');
  });
  test('$isHighlighted and $pseudoSelected', () => {
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $isHighlighted: true,
            $pseudoSelected: true,
          }),
        ),
      ),
    ).toEqual('0 0 0 0');
  });
  test('$isHighlighted', () => {
    expect(
      toStrVal(
        calcRadius(
          getProps({
            $isHighlighted: true,
          }),
        ),
      ),
    ).toEqual('4px 4px 4px 4px');
  });
});
