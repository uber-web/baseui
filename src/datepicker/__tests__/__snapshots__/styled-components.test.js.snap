// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component styled components Basic render: StyledCalendarContainer has correct styles 1`] = `
Object {
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale400",
}
`;

exports[`Component styled components Basic render: StyledCalendarHeader has correct styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.primary",
  "borderRadius": "$theme.borders.radius200 $theme.borders.radius200 0 0",
  "color": "$theme.colors.white",
  "display": "flex",
  "justifyContent": "space-between",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingLeft": "$theme.sizing.scale600",
  "paddingRight": "$theme.sizing.scale600",
  "paddingTop": "$theme.sizing.scale500",
}
`;

exports[`Component styled components Basic render: StyledDay has correct styles 1`] = `
Object {
  ":first-child": Object {
    "borderBottomLeftRadius": "$theme.borders.radius200",
    "borderTopLeftRadius": "$theme.borders.radius200",
  },
  ":last-child": Object {
    "borderBottomRightRadius": "$theme.borders.radius200",
    "borderTopRightRadius": "$theme.borders.radius200",
  },
  "backgroundColor": "transparent",
  "borderBottomLeftRadius": "$theme.borders.radius200",
  "borderBottomRightRadius": "$theme.borders.radius200",
  "borderTopLeftRadius": "$theme.borders.radius200",
  "borderTopRightRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "inherit",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Basic render: StyledMonth has correct styles 1`] = `
Object {
  "display": "inline-block",
}
`;

exports[`Component styled components Basic render: StyledMonthHeader has correct styles 1`] = `
Object {
  "whiteSpace": "no-wrap",
}
`;

exports[`Component styled components Basic render: StyledQuickSelectButtons has correct styles 1`] = `
Object {
  "display": "flex",
  "flexWrap": "wrap",
}
`;

exports[`Component styled components Basic render: StyledQuickSelectContainer has correct styles 1`] = `
Object {
  "marginBottom": "$theme.sizing.scale600",
  "marginLeft": "auto",
  "marginRight": "auto",
  "maxWidth": "296px",
}
`;

exports[`Component styled components Basic render: StyledQuickSelectLabel has correct styles 1`] = `
Object {
  "color": "$theme.colors.mono800",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "marginBottom": "$theme.sizing.scale600",
  "textAlign": "left",
}
`;

exports[`Component styled components Basic render: StyledRoot has correct styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.white",
  "borderRadius": "$theme.borders.radius200",
  "color": "$theme.colors.black",
  "display": "inline-block",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "textAlign": "center",
}
`;

exports[`Component styled components Basic render: StyledWeek has correct styles 1`] = `
Object {
  "display": "flex",
  "marginBottom": "$theme.sizing.scale100",
  "whiteSpace": "no-wrap",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $disabled is set to true 1`] = `
Object {
  ":first-child": Object {
    "borderBottomLeftRadius": "$theme.borders.radius200",
    "borderTopLeftRadius": "$theme.borders.radius200",
  },
  ":last-child": Object {
    "borderBottomRightRadius": "$theme.borders.radius200",
    "borderTopRightRadius": "$theme.borders.radius200",
  },
  "backgroundColor": "transparent",
  "borderBottomLeftRadius": "$theme.borders.radius200",
  "borderBottomRightRadius": "$theme.borders.radius200",
  "borderTopLeftRadius": "$theme.borders.radius200",
  "borderTopRightRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono600",
  "cursor": "default",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $isHighlighted is set to true 1`] = `
Object {
  ":first-child": Object {
    "borderBottomLeftRadius": "$theme.borders.radius200",
    "borderTopLeftRadius": "$theme.borders.radius200",
  },
  ":last-child": Object {
    "borderBottomRightRadius": "$theme.borders.radius200",
    "borderTopRightRadius": "$theme.borders.radius200",
  },
  "backgroundColor": "$theme.colors.primary100",
  "borderBottomLeftRadius": "$theme.borders.radius200",
  "borderBottomRightRadius": "$theme.borders.radius200",
  "borderTopLeftRadius": "$theme.borders.radius200",
  "borderTopRightRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "inherit",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $isHovered is set to true 1`] = `
Object {
  ":first-child": Object {
    "borderBottomLeftRadius": "$theme.borders.radius200",
    "borderTopLeftRadius": "$theme.borders.radius200",
  },
  ":last-child": Object {
    "borderBottomRightRadius": "$theme.borders.radius200",
    "borderTopRightRadius": "$theme.borders.radius200",
  },
  "backgroundColor": "$theme.colors.primary100",
  "borderBottomLeftRadius": "$theme.borders.radius200",
  "borderBottomRightRadius": "$theme.borders.radius200",
  "borderTopLeftRadius": "$theme.borders.radius200",
  "borderTopRightRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "inherit",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $outsideMonth is set to true 1`] = `
Object {
  ":first-child": Object {
    "borderBottomLeftRadius": "$theme.borders.radius200",
    "borderTopLeftRadius": "$theme.borders.radius200",
  },
  ":last-child": Object {
    "borderBottomRightRadius": "$theme.borders.radius200",
    "borderTopRightRadius": "$theme.borders.radius200",
  },
  "backgroundColor": "transparent",
  "borderBottomLeftRadius": "$theme.borders.radius200",
  "borderBottomRightRadius": "$theme.borders.radius200",
  "borderTopLeftRadius": "$theme.borders.radius200",
  "borderTopRightRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono600",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;

exports[`Component styled components Prop based styles: StyledDay has correct styles when $selected is set to true 1`] = `
Object {
  ":first-child": Object {
    "borderBottomLeftRadius": "$theme.borders.radius200",
    "borderTopLeftRadius": "$theme.borders.radius200",
  },
  ":last-child": Object {
    "borderBottomRightRadius": "$theme.borders.radius200",
    "borderTopRightRadius": "$theme.borders.radius200",
  },
  "backgroundColor": "$theme.colors.primary",
  "borderBottomLeftRadius": "$theme.borders.radius200",
  "borderBottomRightRadius": "$theme.borders.radius200",
  "borderTopLeftRadius": "$theme.borders.radius200",
  "borderTopRightRadius": "$theme.borders.radius200",
  "boxSizing": "border-box",
  "color": "$theme.colors.white",
  "cursor": "pointer",
  "display": "inline-block",
  "height": "$theme.sizing.scale1000",
  "lineHeight": "$theme.sizing.scale800",
  "marginBottom": "0",
  "marginLeft": "0",
  "marginRight": "0",
  "marginTop": "0",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale200",
  "paddingRight": "$theme.sizing.scale200",
  "paddingTop": "$theme.sizing.scale300",
  "position": "relative",
  "textAlign": "center",
  "width": "$theme.sizing.scale1000",
}
`;
